### PR TITLE
#483 Remove duplicated code detected by Sonar

### PR DESCRIPTION
**Pull request by rosiereflo**

### CODE REVIEW

File: util/travis_ci/launch_command.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 if [ "$TRAVIS_BRANCH" = "master" -a "$TRAVIS_EVENT_TYPE" = "push" ]; then
-  echo -n "export DYNAWO_BUILD_TYPE=TestCoverage;cd dynawo;export DYNAWO_THIRD_PARTY_INSTALL_DIR=/opt/dynawo/3rParty;/opt/sonar/build-wrapper-linux-x86/build-wrapper-linux-x86-64 --out-dir bw-output ./util/envDynawo.sh build-tests-coverage;RETURN_CODE=\$?;if [ \${RETURN_CODE} -ne 0 ]; then exit \${RETURN_CODE}; fi;/opt/sonar/sonar-scanner-3.3.0.1492-linux/bin/sonar-scanner;RETURN_CODE=\$?;if [ \${RETURN_CODE} -ne 0 ]; then exit \${RETURN_CODE}; fi;util/envDynawo.sh build-dynawo-models;"
+  echo -n "export DYNAWO_BUILD_TYPE=TestCoverage;export DYNAWO_RESULTS_SHOW="false";cd dynawo;export DYNAWO_THIRD_PARTY_INSTALL_DIR=/opt/dynawo/3rParty;/opt/sonar/build-wrapper-linux-x86/build-wrapper-linux-x86-64 --out-dir bw-output ./util/envDynawo.sh build-tests-coverage;RETURN_CODE=\$?;if [ \${RETURN_CODE} -ne 0 ]; then exit \${RETURN_CODE}; fi;/opt/sonar/sonar-scanner-3.3.0.1492-linux/bin/sonar-scanner;RETURN_CODE=\$?;if [ \${RETURN_CODE} -ne 0 ]; then exit \${RETURN_CODE}; fi;util/envDynawo.sh build-dynawo-models;"
```
Comment (gautierbureau) [inline util/travis_ci/launch_command.sh]: the reason why sonar failed ?

File: dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp [hunk 2]
```diff
@@ -95,20 +96,7 @@ ModelTemplate::Impl::addUnitDynamicModel(const shared_ptr<UnitDynamicModel>& mod
...
-  string pc_first = model1 + '_' + var1;
-  string pc_second = model2 + '_' + var2;
-  // To build the connector Id, sort the string so as 1st_Model_ID is smaller than 2nd_Model_ID. EX: ID_1 < ID_2
-  string pc_Id;
-  if (pc_first < pc_second)
-    pc_Id = pc_first + '_' + pc_second;
-  else
-    pc_Id = pc_second + '_' + pc_first;
-
+  string pc_Id = getConnectionId(model1, var1, model2, var2, getId(), unitDynamicModelsMap_);
```
Comment (gautierbureau) [inline dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp]: what is the meaning of pc ?

File: util/travis_ci/launch_command.sh (hunk 1 above)
Comment (rosiereflo) [reply util/travis_ci/launch_command.sh]: Hope so!

File: dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp (hunk 2 above)
Comment (rosiereflo) [reply dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp]: absolutely no idea. I kept the name but I can change it if necessary

File: util/travis_ci/launch_command.sh (hunk 1 above)
Comment (gautierbureau) [reply util/travis_ci/launch_command.sh]: ok

File: dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp (hunk 2 above)
Comment (gautierbureau) [reply dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp]: I saw it is also used in the dictionnary so maybe if we don't someone else might not know... if it has a real meaning...
there is the same with ic below, I presumed init connect but not sure also

Comment (rosiereflo) [reply dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp]: According to Adrien it might be "part" of the model, as it is a connection within the model

Comment (gautierbureau) [reply dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp]: It means Pin Connection

Comment (rosiereflo) [reply dynawo/sources/API/DYD/DYDModelTemplateImpl.cpp]: Solved

**Pull request merged**